constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/ConstraintSetParser.java

```diff
@@ -726,6 +737,114 @@ static void parseChain(int orientation, State state,
+                        Utils.log("================ " + i + " " + chainElement.getClass() + "  "
+                                + chainElement.content());
+                        if (chainElement instanceof CLArray) {
+                            CLArray array = (CLArray) chainElement;
+                            if (array.size() == 1) {
+                                chain.add(array.get(0).content());
+                            } else {
+                                String id = array.get(0).content();
+                                float weight = Float.NaN;
+                                float preMargin = Float.NaN;
+                                float postMargin = Float.NaN;
+
+                                if (array.size() >= 2) {
+                                    weight = array.getFloat(1);
+                                    if (array.size() >= 3) {
+                                        postMargin = preMargin = toPix(state, array.getFloat(2));
+                                        if (array.size() >= 4) {
+                                            postMargin =  toPix(state, array.getFloat(3));
+                                        }
+                                    }
+                                }
+
+                                chain.addChainElement(id, weight, preMargin, postMargin);
+                            }
+
+                        } else {
+                            chain.add(chainElement.content());
+                        }
+                    }
+                    break;
+                case "start":
+                case "end":
+                case "top":
+                case "bottom":
+                case "left":
+                case "right":
+                    parseConstraint(state, margins, object, chain, params);
+                    break;
+                case "style":
+
+                    CLElement styleObject = object.get(params);
+                    String styleValue;
+                    if (styleObject instanceof CLArray && ((CLArray) styleObject).size() > 1) {
+                        styleValue = ((CLArray) styleObject).getString(0);
+                        float biasValue = ((CLArray) styleObject).getFloat(1);
+                        chain.bias(biasValue);
+                    } else {
+                        styleValue = styleObject.content();
+                    }
+                    switch (styleValue) {
+                        case "packed":
+                            chain.style(State.Chain.PACKED);
+                            break;
+                        case "spread_inside":
+                            chain.style(State.Chain.SPREAD_INSIDE);
+                            break;
+                        default:
+                            chain.style(State.Chain.SPREAD);
+                            break;
+                    }
+
+                    break;
+            }
```

Comment:
This is the definition of this behavior.
If we do not understand the string we set it to SPREAD
I would ideally like:
case "spread":
default: 
  chain.style.(State.Chain.SPREAD); 
  
  to be clear what the behavior but that would break a picky -Werror flag